GVFS/GVFS.Platform.Windows/WindowsPlatform.cs

```diff
@@ -285,5 +287,61 @@ private static object GetValueFromRegistry(RegistryHive registryHive, string key
             object value = localKeySub == null ? null : localKeySub.GetValue(valueName);
             return value;
         }
+
+        public override bool OpenAndReadFile(string fileName)
+        {
+            return NativeFileReader.OpenAndReadFile(fileName);
+        }
+
+        private class NativeFileReader
+        {
+            private const uint GenericRead = 0x80000000;
+            private const uint OpenExisting = 3;
+
+            public static bool OpenAndReadFile(string fileName)
+            {
+                using (SafeFileHandle handle = Open(fileName))
+                {
+                    if (!handle.IsInvalid)
+                    {
+                        return ReadOneByte(handle);
+                    }
+                }
+                return false;
+            }
+
+            private static SafeFileHandle Open(string fileName)
+            {
+                return CreateFile(fileName, GenericRead, (uint)(FileShare.ReadWrite | FileShare.Delete), 0, OpenExisting, 0, 0);
+            }
+
+            private static unsafe bool ReadOneByte(SafeFileHandle handle)
+            {
+                byte[] buffer = new byte[1];
+                int n = 0;
+                fixed (byte* p = buffer)
+                {
+                    return ReadFile(handle, p, 1, &n, 0);
+                }
+            }
+
+            [DllImport("kernel32", SetLastError = true, ThrowOnUnmappableChar = true, CharSet = CharSet.Unicode)]
+            private static extern unsafe SafeFileHandle CreateFile(
+                string fileName,
+                uint desiredAccess,
+                uint shareMode,
+                uint securityAttributes,
+                uint creationDisposition,
+                uint flagsAndAttributes,
+                int hemplateFile);
+
+            [DllImport("kernel32", SetLastError = true)]
+            private static extern unsafe bool ReadFile(
+                SafeFileHandle file,
+                void* buffer,

```

Comment:
Can we use `[Out] byte []` here as well and avoid having to add `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>`?

Reply:
FYI we need to use fixed not just for the type conversion, but also to ensure the GC doesn't move the memory. So we're stuck with the unsafe keywords here.